subsys/bluetooth/host/classic/a2dp.c

```
@@ -76,35 +173,713 @@ struct bt_a2dp *get_new_connection(struct bt_conn *conn)
+			a2dp->peer_seids[a2dp->peer_seids_count] = *seid;
+			a2dp->peer_seids_count++;
+			seid = bt_avdtp_parse_seids(DISCOVER_REQ(req)->buf);
+		}
+
+		/* trigger the getting capability */
+		a2dp->get_capabilities_index = 0xFFu;
+		if (bt_a2dp_get_seid_caps(a2dp) != 0) {
+			/* the peer_seids' caps is done.*/
+			if (a2dp->auto_configure_enabled) {
+				bt_a2dp_select_and_configure(a2dp);
+			} else if (a2dp->discover_endpoint_cb != NULL) {
+				a2dp->discover_endpoint_cb(a2dp, NULL, 0);
+			}
+		}
+	} else {
+		if (a2dp->auto_configure_enabled) {
+			bt_a2dp_select_and_configure(a2dp);
+		} else if (a2dp->discover_endpoint_cb != NULL) {
+			a2dp->discover_endpoint_cb(a2dp, NULL, -1);
+		}
+	}
+
+	return 0;
+}
+
+int bt_a2dp_configure(struct bt_a2dp *a2dp, void (*result_cb)(int err))
+{
+	int err;
+
+	if (a2dp == NULL) {
+		return -EINVAL;
+	}
+
+	if (a2dp->a2dp_state != INTERNAL_STATE_AVDTP_CONNECTED) {
+		return -EIO;
+	}
+
+	a2dp->auto_select_endpoint_index = 0xFFu;
+	a2dp->auto_configure_enabled = 1U;
+	a2dp->configure_cb = result_cb;
+
+	a2dp->discover_param.req.func = bt_a2dp_discover_cb;
+	a2dp->discover_param.buf = NULL;
+	err = bt_avdtp_discover(&a2dp->session, &a2dp->discover_param);
+	if (err) {
+		LOG_DBG("AVDTP discover failed");
+	}
+	return 0;
+}
+
+int bt_a2dp_discover_peer_endpoints(struct bt_a2dp *a2dp, bt_a2dp_discover_peer_endpoint_cb_t cb)
+{
+	int err;
+
+	if (a2dp == NULL) {
+		return -EINVAL;
+	}
+
+	a2dp->auto_configure_enabled = 0U;
+	a2dp->discover_endpoint_cb = cb;
+	a2dp->discover_param.req.func = bt_a2dp_discover_cb;
+	a2dp->discover_param.buf = NULL;
+	err = bt_avdtp_discover(&a2dp->session, &a2dp->discover_param);
+	if (err) {
+		LOG_DBG("AVDTP discover failed");
+	}
+
+	return err;
+}
+
+int bt_a2dp_configure_endpoint(struct bt_a2dp *a2dp, struct bt_a2dp_endpoint *endpoint,
+		struct bt_a2dp_endpoint *peer_endpoint,
+		struct bt_a2dp_endpoint_config *config)
+{
+	/* todo: see the API description in a2dp.h */
+	return 0;
```

It is better to return an error code if the feature is unsupported.